### PR TITLE
Watersystem frontend

### PR DESCRIPTION
**Pull request by gusteran**
- Add more information about the water systems that gets pulled into the map overlay
- Add information about contaminants upon clicking their name
- Add a new icon for the superfund sites
- Refactor database to use Text instead of Strings for Entity properties of encoded objects
- Add header to map page and fix map height to account for the header

Screenshot 2020-07-27 at 6 12 27 PM

### CODE REVIEW

**Comment by gusteran** [comment]
- Add some CSS tweaks to visuals
- Add violation dates and enforcement action for contaminants
- Limit overlay height and introduce scrollbar
- 

image

File: src/main/java/com/google/sps/data/WaterSystem.java [hunk 1]
```diff
@@ -126,7 +127,7 @@ public WaterSystem(Entity entity) throws EntityNotFoundException, IOException {
         this.county = (String) entity.getProperty(COUNTY_PROPERTY);
         this.populationServed = ((Long) entity.getProperty(POPULATION_PROPERTY)).intValue();
         this.contaminants = new ArrayList<WaterContaminant>();
-        byte[] contaminantsData = Base64.getDecoder().decode((String)entity.getProperty(CONTAMINANTS_PROPERTY));
+        byte[] contaminantsData = Base64.getDecoder().decode(((Text)entity.getProperty(CONTAMINANTS_PROPERTY)).getValue());
```
Comment (lazau-g) [inline src/main/java/com/google/sps/data/WaterSystem.java:130]: nit: You're on a good track describing the high level changes in your PR and commit meesages. One way to improve the quality of these messages though is to explain why the changes are being made. Here, you can expand on why you're switching from String to Text.

Given this project is close to being complete you don't really have to do this here, but it's good to know for production projects.

File: src/main/webapp/map.js [hunk 2]
```diff
@@ -153,12 +156,45 @@ function addWaterSystem(areaType, zipCode){
             console.log(system);
             console.log(system.contaminants);
             waterPollutionHTML += 
-                "<h2>"+system.name+"</h2>"+
-                "<h4>Serves "+system.populationServed+"</h4>";
+                "<div id='"+system.pwsid+"' class='water_pollution'>"+
```
Comment (lazau-g) [inline src/main/webapp/map.js:159]: nit: Again, no action required, but something for you to learn. Usually you don't hard code HTML into JS , but rather have template files and use a templating engine to generate the final HTML to inject into a page. See https://github.com/janl/mustache.js for an example.

File: src/main/java/com/google/sps/data/WaterSystem.java (hunk 1 above)
Comment (gusteran) [reply src/main/java/com/google/sps/data/WaterSystem.java:130]: Understood, in this case I'm switching from String to Text because String properties have a size limit whereas Text properties do not

File: src/main/webapp/map.js (hunk 2 above)
Comment (gusteran) [reply src/main/webapp/map.js:159]: That makes a lot of sense, it felt strange to write it like that
